tests/core/test_pandas_engine.py

```diff
@@ -16,7 +16,13 @@
 
 
 @pytest.mark.parametrize(
-    "data_type", list(pandas_engine.Engine.get_registered_dtypes())
+    "data_type",
+    [
+        data_type
+        for data_type in pandas_engine.Engine.get_registered_dtypes()
+        if data_type
+        != pandas_engine.ArrowString  # `string[pyarrow]` gets parsed to `string` by pandas
```

Comment:
one strategy here is to do a check in the `test_pandas_data_type` function body and `pytest.skip` if pandas > 2 and pyarrow is installed.

Reply:
i fixed it like this, i see this pattern in some other tests.
```py
UNSUPPORTED_DTYPE_CLS: set[Any] = set()

# `string[pyarrow]` gets parsed to type `string` by pandas
if pandas_engine.PYARROW_INSTALLED and pandas_engine.PANDAS_2_0_0_PLUS:
    UNSUPPORTED_DTYPE_CLS.add(pandas_engine.ArrowString)


@pytest.mark.parametrize(
    "data_type",
    [
        data_type
        for data_type in pandas_engine.Engine.get_registered_dtypes()
        if data_type not in UNSUPPORTED_DTYPE_CLS
    ],
)
def test_pandas_data_type(data_type):
    ...
```

I just want to make sure it passes all the ci checks :D

Reply:
lgtm